### ISSUE TITLE
Writeable docs should be clearer

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#2509]
The Writeable docs show up in most of our `Formatted*` types. They currently say "writes bytes to a sink", which is not very clear to developers. More clear would be "writes string to a sink".

Writeable has a non-optional `write_to_string` function, which suggests that the bytes should always be UTF-8, right?

@robertbastian

**Comment by robertbastian** [comment]
Yes, writeable is inherently UTF-8, the sinks are not `std::io::Write` but `std::fmt::Write`
